### PR TITLE
Added ability to separate chat for teams

### PR DESCRIPTION
**Pull request by F0RIS**

### CODE REVIEW

**Comment by MastaCoder** [comment]
Can someone test this ^^

**Comment by AlexHGaming** [comment]
@MastaCoder I'll do it.

File: src/GameServer.js [hunk 1]
```diff
@@ -516,8 +517,16 @@ GameServer.prototype.checkBadWord = function(value) {
 GameServer.prototype.sendChatMessage = function(from, to, message) {
     for (var i = 0, len = this.clients.length; i < len; i++) {
         if (!this.clients[i]) continue;
-        if (!to || to == this.clients[i].playerTracker)
-            this.clients[i].packetHandler.sendPacket(new Packet.ChatMessage(from, message));
+        if (!to || to == this.clients[i].playerTracker){
+            if (this.config.separateChatForTeams && this.gameMode.haveTeams){
+                if (from == null /*server*/ || from.team === this.clients[i].playerTracker.team){
```
Comment (MastaCoder) [inline src/GameServer.js]: Remove new line from above an remove comment from /* - You can put this as a new line above. :)

**Comment by AlexHGaming** [comment]
I have no idea on how to trigger it lol

File: src/GameServer.js (hunk 1 above)
Comment (F0RIS) [inline src/GameServer.js]: I didn't get what you want. Put all checks in one if?

Comment (MastaCoder) [inline src/GameServer.js]: Ah nvm I'll fix it if it's merged xD

Comment (F0RIS) [inline src/GameServer.js]: oh, I get, u want make something like
```
// from equals null if message from server
if (from == null || from.team === this.clients[i].playerTracker.team){
...
```

Comment (MastaCoder) [inline src/GameServer.js]: Yeah!

**Comment by F0RIS** [comment]
@AlexHGaming 
https://github.com/Megabyte918/MultiOgar-Edited/pull/720/files#diff-d2ebb6a4d6e9176c1d844ddaad8bf9bdR65

**Comment by MastaCoder** [comment]
I get what this, thanks!

**Comment by F0RIS** [comment]
Hm, probably I should write description next time. I thought that from title everything will be obviously)

**Comment by Agarian** [comment]
Nice work!

**Pull request merged**